packages/amplify_api/lib/src/graphql/graphql_request_factory.dart

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_api/amplify_api.dart';
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+import 'package:flutter/foundation.dart';
+
+class GraphQLRequestFactory {
+  String _getModelType(ModelFieldTypeEnum? val) {
+    switch (val) {
+      case ModelFieldTypeEnum.string:
+        return "String";
+      case ModelFieldTypeEnum.int:
+        return "int";
+      case ModelFieldTypeEnum.model:
+        return "ID";
+      default:
+        return "Error: UnknownType!";
+    }
+  }
+
+  String _getFieldsFromModelType(ModelSchema schema) {
+    Map<String, ModelField> fieldsMap = schema.fields!;
+    return fieldsMap.entries
+        .map((entry) => entry.value.association == null ? entry.key : '')
+        .toList()
+        .join(' ');
+  }
+
+  ModelSchema _getAndValidateSchema(
+      ModelType modelType, GraphQLRequestOperation operation) {
+    ModelProviderInterface? provider = AmplifyAPI.instance.getModelProvider();
+
+    if (provider == null) {
+      throw ApiException('No modelProvider found',
+          recoverySuggestion:
+              'Pass in a modelProvider instance while instantiating APIPlugin');
+    }
+
+    ModelSchema schema = provider.modelSchemas.firstWhere(
+        (elem) => provider.getModelTypeByModelName(elem.name) == modelType,
+        orElse: () => throw ApiException(
+            'No schema found for the ModelType provided',
+            recoverySuggestion:
+                'Pass in a valid modelProvider instance while instantiating APIPlugin or provide a valid ModelType'));
+
+    if (schema.fields == null) {
+      throw ApiException('Schema found does not have a fields property',
+          recoverySuggestion:
+              'Pass in a valid modelProvider instance while instantiating APIPlugin');
+    }
+
+    if (operation == GraphQLRequestOperation.list &&
+        schema.pluralName == null) {
+      throw ApiException('No schema name found',
+          recoverySuggestion:
+              'Pass in a valid modelProvider instance while instantiating APIPlugin or provide a valid ModelType');
+    }
+
+    return schema;
+  }
+
+  /*
```

Comment:
nit: suggest changing this block comment to 
```
/**
   *  Example output `GraphQLRequest` with document string: 
   *    `"query getBlog($id: ID!) { getBlog(id: $id) { id name createdAt } }"`
  */
```
I believe the 2 stars make it a JSdoc style block comment, which also seems to work in dart so the comment will document the method properly. Later, might want to think about documenting the public APIs, but not needed her IMO>